velox/connectors/clp/tests/ClpConnectorTest.cpp

```diff
@@ -0,0 +1,311 @@
+  const std::string kClpConnectorId = "test-clp";
+
+  void SetUp() override {
+    OperatorTestBase::SetUp();
+    connector::registerConnectorFactory(
+        std::make_shared<connector::clp::ClpConnectorFactory>());
+    auto clpConnector =
+        connector::getConnectorFactory(
+            connector::clp::ClpConnectorFactory::kClpConnectorName)
+            ->newConnector(
+                kClpConnectorId,
+                std::make_shared<config::ConfigBase>(
+                    std::unordered_map<std::string, std::string>{
+                        {"clp.split-source", "local"}}));
+    connector::registerConnector(clpConnector);
+  }
+
+  void TearDown() override {
+    connector::unregisterConnector(kClpConnectorId);
+    connector::unregisterConnectorFactory(
+        connector::clp::ClpConnectorFactory::kClpConnectorName);
+    OperatorTestBase::TearDown();
+  }
+
+  exec::Split makeClpSplit(const std::string& splitPath) {
+    return exec::Split(
+        std::make_shared<ClpConnectorSplit>(kClpConnectorId, splitPath));
+  }
+
+  RowVectorPtr getResults(
+      const core::PlanNodePtr& planNode,
+      std::vector<exec::Split>&& splits) {
+    return exec::test::AssertQueryBuilder(planNode)
+        .splits(std::move(splits))
+        .copyResults(pool());
+  }
+
+  static std::string getExampleFilePath(const std::string& filePath) {
+    std::string current_path = fs::current_path().string();
+    return current_path + "/examples/" + filePath;
+  }
+};
+
+TEST_F(ClpConnectorTest, test1NoPushdown) {
+  auto plan = PlanBuilder()
+                  .startTableScan()
+                  .outputType(
+                      ROW({"requestId", "userId", "method"},
+                          {VARCHAR(), VARCHAR(), VARCHAR()}))
+                  .tableHandle(std::make_shared<ClpTableHandle>(
+                      kClpConnectorId, "test_1", nullptr))
+                  .assignments({
+                      {"requestId",
+                       std::make_shared<ClpColumnHandle>(
+                           "requestId", "requestId", VARCHAR(), true)},
+                      {"userId",
+                       std::make_shared<ClpColumnHandle>(
+                           "userId", "userId", VARCHAR(), true)},
+                      {"method",
+                       std::make_shared<ClpColumnHandle>(
+                           "method", "method", VARCHAR(), true)},
+                  })
+                  .endTableScan()
+                  .filter("method = 'GET'")
+                  .planNode();
+
+  auto output =
+      getResults(plan, {makeClpSplit(getExampleFilePath("test_1.clps"))});
+  auto expected = makeRowVector(
+      {// requestId
+       makeFlatVector<StringView>(
+           {"req-100", "req-105", "req-107", "req-109", "req-102"}),
+       // userId
+       makeNullableFlatVector<StringView>(
+           {"user201", "user204", "user202", "user203", std::nullopt}),
+       // method
+       makeFlatVector<StringView>({
```

Comment:
```suggestion
       // requestId
       makeFlatVector<StringView>(
           {"req-100", "req-105", "req-107", "req-109", "req-102"}),

       // userId
       makeNullableFlatVector<StringView>(
           {"user201", "user204", "user202", "user203", std::nullopt}),

       // method
       makeFlatVector<StringView>({
```

Will the formatter allow this? If so, can you apply it in the other places?

Reply:
An extra newline is allowed, but after formatting, the `{` should remain on the same line as `// requestId`.

Reply:
Do you think we should add this extra new line?

Reply:
Yes, if you could, just to stay consistent with our own coding standards.

Reply:
which rule you are talking about? And do you want me to apply this to all other comments？

Reply:
Seems a bit weird to add a new line for other places

Reply:
> which rule you are talking about? And do you want me to apply this to all other comments？

https://www.notion.so/yscope/WIP-Coding-Guidelines-9a308b847a5343958ba3cb97a850be66?source=copy_link#14a04e4d9e6b801dac69d6222e8885a2

> Include an empty line before every comment block.

> Seems a bit weird to add a new line for other places

Which other places?

Reply:
For example
```
       makeRowVector({// event.type
                      makeFlatVector<StringView>({"storage"}),

                      // event.subtype
                      makeFlatVector<StringView>({"disk_usage"}),

                      // event.severity
                      makeFlatVector<StringView>({"WARNING"}),

                      // event.tags
                      makeArrayVector<StringView>(
                          {{"\"filesystem\"", "\"monitoring\""}})})
```
This extra new line seems a bit weird.
And because they are parameters, not normal code statements, should we follow this rule?

Reply:
Mm, because of the way their formatting is configured, I guess we can ignore it in these cases. But let's stick to it in our codebase since it will be easier to automate that way.

Reply:
So the conclusion is that we don't need to apply it in this connector test, but instead apply it in the CLP or another YScope repository?

Reply:
Correct